ledger/participant-integration-api/src/main/resources/db/migration/h2database/V31__rejection_status_proto.sql

```diff
@@ -0,0 +1,16 @@
+--  Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+--  SPDX-License-Identifier: Apache-2.0
+
+-- Store the completion status as a serialized Protocol Buffers message of type `google.rpc.Status`.
+
+-- The `rejection_status` contains a Protocol-Buffers-serialized message of type
+-- `google.rpc.Status`, containing the code, message, and further details (decided by the ledger
+-- driver). The `rejection_status_code` and `rejection_status_message` columns may contain old
+-- data and will always be `NULL` in a H2 ledger.
```

Comment:
> ...will always be `NULL` in a H2 ledger.

Why is that?

Reply:
Because it's in memory, so all migrations are always run on an empty ledger.

Reply:
I can vaguely remember we also it use on-disk (but, if so, perhaps with sandbox) but I may be confused.

Reply:
We _can_, but it's not officially supported and you should not do that.

Using the append-only index, we do not support migrations at all on H2 (which is why I modified it directly).